File: gameFramework/src/main/java/at/grueneis/game/framework/code/AndroidFileIO.java
package at.grueneis.game.framework.code;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

import android.content.res.AssetManager;
import android.os.Environment;
import at.grueneis.game.framework.FileIO;

public class AndroidFileIO implements FileIO
{
	AssetManager assets;
	String sdPath;
	
	public AndroidFileIO(AssetManager assets)
	{
		this.assets = assets;
		sdPath = Environment.getExternalStorageDirectory().getAbsolutePath() + File.separator;
	}
	
	public InputStream readAsset(String filename) throws IOException
	{
		return assets.open(filename);
	}
	
	public InputStream readFile(String filename) throws IOException
	{
		return new FileInputStream(sdPath + filename);
	}
	
	public OutputStream writeFile(String filename) throws IOException
	{
		return new FileOutputStream(sdPath + filename);
	}
	
}

File: game-framework/src/main/kotlin/at/grueneis/game/framework/impl/AndroidFileIO.kt
package at.grueneis.game.framework.impl

import android.content.res.AssetManager
import android.os.Environment
import at.grueneis.game.framework.FileIO
import java.io.*

class AndroidFileIO(val assets: AssetManager) : FileIO {
    var sdPath: String = Environment.getExternalStorageDirectory().absolutePath + File.separator

    @Throws(IOException::class)
    override fun readAsset(filename: String) = assets.open(filename)

    @Throws(IOException::class)
    override fun readFile(filename: String): InputStream {
        return FileInputStream(sdPath + filename)
    }

    @Throws(IOException::class)
    override fun writeFile(filename: String): OutputStream {
        return FileOutputStream(sdPath + filename)
    }
}